### ISSUE TITLE
DustTransfer: "You don't have this asset."

### ISSUE THREAD

**Issue by anotherdeveloper** [JKorf/Binance.Net#282]
DustTransfer always returns "You don't have this asset.", including when the Binance website allows me to convert dust in that asset to BNB.

**Comment by JKorf** [comment]
I don't have this issue, just tried it and converted successfully. How are you calling it and do you have some debug logging?

**Comment by anotherdeveloper** [comment]
Logging had the answer, and the answer was that I had done something stupid. Thanks, and sorry to have bothered you with this.

**Comment by JKorf** [comment]
All good

**Comment by anotherdeveloper** [comment]
I think I was wrong, and I think that there's a problem with serialization of parameters in DustTransferAsync. Here's the URL it generates - I removed irrelevant stuff from end:


```
https://api.binance.com/sapi/v1/asset/dust?asset=System.Linq.Enumerable+WhereSelectEnumerableIterator`2[<>f__AnonymousType2`2[<>f__AnonymousType0`2[Binance.Net.Objects.BinanceBalance,TradingHelpers.BtcRel],System.Decimal],System.String]
```

The assets I was passing in to DustTransferAsync was an `IEnumerable<string>` that was the result of a Linq query. The fix (on my end) was to call ToArray() on this before calling DustTransferAsync. 

ToList() doesn't work.

**Comment by JKorf** [comment]
You're right, the data serialization fails when not using an array. I tested with an array, so I didn't notice.

Thanks for the update, I've pushed a fix in 5.0.9 which should be online in a few minutes.
